app/controllers/games_controller.rb

```diff
@@ -19,7 +19,48 @@ def show
 
   def update
     @game = Game.find(params[:id])
-    @game.update_attributes(game_params)
+
+    if request.xhr?
+      # e.g. pawnBlack or kingWhite, etc
+      selected_piece = request['selected_piece']
+
+      # Square that the selected_piece initially
+      # resides in. Format is row4col4, or row1col6, etc
+      initial_square = request['initial_square']
+
+      # Square that the user wants the selected piece
+      # to move to. Format is row4col4, or row1col6, etc
+      final_square = request['final_square']
+
+      # Create backend logic in models and call them here
+      # with the incoming data as arguments
+
+      # data to be returned. Will contain the return of
+      # the model logic.
+      # Piece_captured is so javascript can update the
+      # captured pieces board.
+      # Message would go in the game status board, so
+      # stuff like 'your turn' or 'invalid move',
+      # 'checkmate', etc.
+      valid_move = true
+      piece_captured = 'pawnWhite'
+      message = nil
+
+      # feel free to add new data entries to the output
+      respond_to do |format|
+        format.json do
+          render json: {
+            validMove: valid_move,
+            pieceCaptured: piece_captured,
+            message: nil
+          }.to_json
+        end
+      end
+
+    else
+      # This code runs if the request is not ajax
+      @game.update_attributes(game_params)
+    end
```

Comment:
So create a pieces_controller, hook up an update/#patch route, and then send the ajax request to there. In the def update we'll call backend logic to convert the incoming data to data that can be used for the backend verifications, do the verifications, and then send the data back in the response. The one thing I'm not sure about is what url to send ajax to. I currently have it so there's a hidden div that grabs the id of the current game in show.html, and then in the ajax request, i just get that data before the request and append it to /games/ (so like /games/1). I'm not really sure how to do that with the piece though. The response also contains data, such as the message which pertains to the game. Should the message be directed to piece.game_id.black_player_id/white_player_id? Not really sure how to do that D: